In [2]:
import matplotlib.pyplot as plt
import scipy.stats as st
import seaborn as sns
import pandas as pd
import numpy as np

## Wichtigste Übungsaufgaben:
* 4.1 
  * a
  * b
* 4.3
* 4.4
* 4.5
---
# Aufgabe 4.1
### a) Gegeben sind zwei unabhängige Zufallsvariable X und Y mit den Kennwerten $μ_X = 40$, $\sigma_X = 15$, $μ_Y = 85$ und $\sigma_Y = 18$. Berechnen Sie $E(X +2Y)$, $Var(X +2Y)$ und $E(X^2)$.

Ich schliesse auf:<br>
<center>$\mu_x = E(X) = 40$ & $\sigma_X = 15$ => $VAR(X) = \sigma^2_X = 15^2$</center>
<center>$\mu_y = E(Y) = 85$ & $\sigma_Y = 18$ => $VAR(Y) = \sigma^2_Y = 18^2$</center>
<br>
Berechnung $E(X^2)$ 
$$ VAR(X) = E(X^2) - E(X)^2$$
$$ 15^2 = E(X^2) - 40^2$$
$$E(X^2) = 1825$$
<br>
Berechnung $E(X+2Y)$ <br />
$$E(X+2Y) = E(X) + E(2Y)$$
$$E(X+2Y) = 40 + 2*85$$
$$E(X+2Y) = 210$$
<br>
Berechnung $VAR(X+2Y)$ <br />
$$VAR(X+2Y) = VAR(a+bX) = b^2*VAR(X)$$
<br />
$$a = X = VAR(X) = 15^2$$
$$b = 2 == b^2 = 2^2 = 4$$
$$(X) = (Y) = VAR(Y) = 18^2$$
<br />
$$VAR(X+2Y) = VAR(X) + 2^2*VAR(Y)$$
$$15^2 + 4 * 18^2 = 1521$$


### b) Ein Werk produziert rechteckige Glasscheiben, deren Länge X und Breite Y (in mm gemessen) voneinander unabhängig produktionsbedingten Schwankungen unterliegen. Es gilt $μ_X = 1000$, $s_X = 0.02$, $μ_Y = 500$, $s_Y = 0.01$. Wie gross sind Erwartungswert und Standardabweichung des Umfangs U?

$$U = 2*l + 2*b = 2(l+b)$$<br>
$E(S_n) = n\mu$ n in unserem Fall 2 (je einmal für Länge und dann gleiches Bsp. für Breite
$$E(S_{nX}) = n\mu = 2 * 1000 = 2000$$
$$E(S_{nY}) = n\mu = 2 * 500 = 1000$$
$$E(S_U) = E(S_{nX}) + E(S_{nY}) = 3000$$<br>
Gleiches Beispiel für $\sigma$
$$\sigma(S_n) = \sqrt{n}*\sigma_x$$
$$\sigma(S_nX) = \sqrt{2}*0.02 = 0.02828$$
$$\sigma(S_nY) = \sqrt{2}*0.01 = 0.01414$$
$$\sigma(S_U) = \sigma(S_nX) + \sigma(S_nY) = 0.04242$$

### c) (Zusatzaufgabe) Bestimmen Sie die Wahrscheinlichkeitsdichte fX(x) der Zufallsvariablen X = Z2, wobei Z ∼ N(0, 1).

---
# Aufgabe 4.2
Erzeugen Siemit norm.rvs()m = 500 Stichproben aus einer Standardnormalverteilung mit Umfang n = 5. Speichern Sie die Stichproben als (n × m)− Matrix ab. <br>Python-Hinweis:<br>
`import matplotlib.pyplot as plt
from scipy.stats import norm
import numpy as np
n = 5
m = 500
ran = np.array(norm.rvs(size=n*m))
sim = ran.reshape((n,m))`
### a) Stellen Sie die Stichproben als Runs graphisch dar, und staunen Sie über die zahlreichen möglichen Verläufe.
`plt.plot(sim)
plt.show()`

### b) Berechnen Sie nun für jede Stichprobe den Mittelwert x, und erzeugen Sie ein Histogramm der Mittelwerte. Aus der Theorie wissen Sie, dass die Mittelwerte normalverteilt mit Parameter μ = 0 und Standardabweichung s = 1 √n sein müssten. Überprüfen Sie dies graphisch mit Hilfe von Histogrammen und der theoretischen Wahrscheinlichkeitsdichtekurve
`plt.hist(sim.T, bins=20, density=True, edgecolor="black",
facecolor="white")
x = np.linspace(-4, 4, num=100)
y = norm.pdf(x)
plt.plot(x,y)
plt.title("Histogramm sim")
plt.show()
sim_mean = sim.mean(axis=0)
plt.hist(sim_mean, density=True, edgecolor="black",
facecolor="white")
x = np.linspace(-4, 4, num=100)
y = norm.pdf(x, loc=0, scale=1/np.sqrt(n))
plt.plot(x,y)
plt.title("Histogramm sim_mean")
plt.show()`

### c) Wiederholen Sie die Aufgabe für n = 2, 10, 100.

# Aufgabe 4.3
### a) Eine Elektronik-Firma stellt Widerstände her, die einen mittleren Widerstand von 100W und eine Standardabweichung von 10W haben. Die Widerstände sind normalverteilt. Bestimmen Sie die Wahrscheinlichkeit, dass sich für eine zufällige Stichprobe von n = 25 Widerständen ein mittlerer Widerstand unter 95W ergibt.

$$X_n = 100\Omega = E(X)$$
$$\sigma_x = 10\Omega$$
$$n = 25$$
$$N(\mu, \frac{\sigma^2}{\sqrt{n}}) = N(100,4)$$
<br />
$$P(X < 95)$$

In [6]:
st.norm.cdf(x=95, loc=100, scale=2)

0.006209665325776132

Die W'keit einen mittleren Widerstand unter 95W zu erhalten bei 25 zufälligen Stichproben beträgt 0.6%

### b) Nehmen Sie an, dass eine Zufallsvariable X einer uniformen Verteilung folgt, und zwar X ∼ Uniform[4,6]. Die Zufallsvariable X40 sei der Mittelwert einer i.i.d. Stichprobe vom Umfang n = 40. Wie lautet die Verteilung von X40 approximativ?

$$n=40$$
gem. Formeln der Uniformen Verteilung
$$\mu = 5$$
$$VAR(X) = \frac{1}{3}$$

Zentraler Grenzwertsatz-Formel
$$X_n = N(\mu, \frac{\sigma^2_x}{n}) = N(5,\frac{\frac{1}{3}}{40}) = N(5,\frac{1}{120})$$

---
# Aufgabe 4.4
Es seien {Xi}1≤i≤50 unabhängige und normalverteilte Zufallsvariablen mit Erwartungswert μ = 1 und Standardabweichung s = 2. Darüber hinaus sind folgende Zufallsvariablen definiert:
$$S_n = X_1 + X_2 + ... + X_n$$
<center>und</center>
<center>arithm. Mittel: $X_n = \frac{1}{n}(X_1 + X_2 + ... + X_n) = \frac{S_n}{n}$</center>
Dabei ist n = 50.

### a) Bestimmen Sie die Parameter der Normalverteilung von $S_n$ sowie $X_n$.

zentraler Grenzwertsatz
$$S_n = N(50, 200)$$
$$X_n = N(1, 0.08)$$

### b) Berechnen Sie dieWahrscheinlichkeit P (E[X1] − 1 ≤ X1 ≤ E[X1] + 1).


In [7]:
st.norm.cdf(x=2, loc=1, scale=2) - st.norm.cdf(x=0, loc=1, scale=2)

0.38292492254802624

Die W'keit beträgt 38%

###  c) Berechnen Sie P (E[Sn] − 1 ≤ Sn ≤ E[Sn] + 1).


In [10]:
st.norm.cdf(x=51, loc=50, scale=np.sqrt(200)) - st.norm.cdf(x=49, loc=50, scale=np.sqrt(200))

0.05637197779701664

Die W'keit beträgt 56%

### d) Berechnen Sie P 􀀀E[Xn] − 1 ≤ Xn ≤ E[Xn] + 1.

In [15]:
st.norm.cdf(x=2, loc=1, scale=np.sqrt(0.08)) - st.norm.cdf(x=0, loc=1, scale=np.sqrt(0.08))

0.999593047982555

In [12]:
st.norm.cdf(x=2, loc=1, scale=np.sqrt(4/50)) - st.norm.cdf(x=0, loc=1, scale=np.sqrt(4/50))

0.999593047982555

---
# Aufgabe 4.5
Wir betrachten einen Datensatz, bei welchem zwei Methoden zur Bestimmung der latenten Schmelzwärme von Eis verglichen werden. Wiederholte Messungen der freigesetzten Wärme beim Übergang von Eis bei −0.7◦C zu Wasser bei 0◦C ergaben die folgenden Werte (in cal/g):

### a) Berechnen Sie die arithmetischen Mittelwerte der beiden Methoden und geben Sie für jede Methode den absoluten Fehler an.

In [17]:
mA = np.array([79.98, 80.04, 80.02, 80.04, 80.03, 80.03, 80.04, 79.97, 80.05, 80.03, 80.02, 80.00, 80.02])
mB = np.array([80.02, 79.94, 79.98, 79.97, 79.97, 80.03, 79.95, 79.97])

In [18]:
mA.mean()

80.02076923076923

In [19]:
mB.mean()

79.97874999999999

absoluter Fehler: arithm.Mittel +- $\frac{standardabweichung}{\sqrt{n}}$

In [21]:
aFmA = mA.std()/np.sqrt(mA.size)
print(aFmA)

0.006386147958926608


absoluter Fehler der Methode A: (80.02 +- 0.01) cal/g

In [23]:
aFmB = mB.std()/np.sqrt(mB.size)
print(aFmB)

0.010373870420436046


absoluter Fehler der Methode B: (79.98 +- 0.01) cal/g

### b) Wie lauten die relativen Fehler?

relativer Fehler: $\frac{absol. Fehler}{arithm. Mittel}$

In [31]:
rFmA = aFmA / mA.mean()
print(rFmA)
'{:f}'.format(rFmA)

7.980613058729551e-05


'0.000080'

rel. Fehler: 80.02cal/g +- 0.000080%

In [32]:
rFmB = aFmB / mB.mean()
print(rFmB)
'{:f}'.format(rFmB)

0.00012970783389882996


'0.000130'

rel. Fehler: 79.98cal/g +- 0.000130%

---
# Aufgabe 4.6
Aufgabenstellung siehe Skript

Berechnen Sie folgendes Integral
$$ I(f) = \frac{1}{\sqrt{2\pi}} \int_0^1e^{-x^2 /2} dx$$
Berechnen Sie das Integral, indem Sie 1000 uniform über das Intervall [0,1] verteilte Zahlen X1, . . . , X1000 mit der Python -Funktion uniform.rvs(...) generieren. Berechnen Sie den genauen numerischenWert des Integralsmit der Python -Funktion norm.cdf(...).